go/vt/schemadiff/table.go

```diff
@@ -1087,10 +1111,10 @@ func (c *CreateTableEntity) diffOptions(alterTable *sqlparser.AlterTable,
 				case "CHARSET", "COLLATE":
 					switch hints.TableCharsetCollateStrategy {
 					case TableCharsetCollateStrict:
-						alterTableOptions = append(alterTableOptions, t2Option)
+						modifyTableOption(t1Option, t2Option)
 					case TableCharsetCollateIgnoreEmpty:
 						if t1Option.String != "" && t2Option.String != "" {
-							alterTableOptions = append(alterTableOptions, t2Option)
+							modifyTableOption(t1Option, t2Option)
```

Comment:
Looks like this isn't covered by the unit test.

Reply:
Added,